src/FormLayout/Generic.php

```diff
@@ -35,43 +36,67 @@ class Generic extends View
     public $inline = null;
 
     /**
-     * Places field inside a layout somewhere.
+     * Places field inside a layout somewhere. Should be called
+     * through $form->addField().
      *
-     * @param \atk4\ui\FormField\Generic|array $field
-     * @param array|string                     $args
+     * @param string|null              $name
+     * @param array|string|object|null $decorator
+     * @param array|string|object|null $field
      *
      * @return \atk4\ui\FormField\Generic
      */
-    public function addField($field, $args = [])
+    public function addField(string $name, $decorator = null, $field = null)
     {
-        if (is_string($args)) {
-            $args = ['caption' => $args];
-        } elseif (is_array($args) && isset($args[0])) {
-            $args['caption'] = $args[0];
-            unset($args[0]);
+        if (!is_string($name)) {
```

Comment:
it's already defined in function definition addField(**string** $name, ... so isit really necessary to check that here too?

Reply:
yes, since user can call $layout->addField too.